File: onto-code-maker-result-tester/src/main/java/ontology/generator/classes/examples/family1/serialization/DogSerialization.java

package ontology.generator.classes.examples.family1.serialization;

import org.eclipse.rdf4j.model.*;
import java.util.Collection;
import java.util.Set;
import java.util.HashSet;
import java.util.*;

import java.util.stream.Collectors;
import ontology.generator.classes.examples.family1.entities.*;
import org.eclipse.rdf4j.model.util.Values;
import org.eclipse.rdf4j.model.vocabulary.RDF;

import java.time.LocalTime;
import ontology.generator.classes.examples.family1.Vocabulary;

public class DogSerialization extends SerializationModel<Dog>{

    @Override
    public void addToModel(Model model, Dog dog) {
        model.add(dog.getIri(),RDF.TYPE, dog.getClassIRI());

    }


    protected void setProperties(Model model,Dog dog,int nestingLevel) throws Exception{

    }

    @Override
    public Dog getInstanceFromModel(Model model,IRI instanceIri,int nestingLevel) throws Exception{
        Model statements = model.filter(instanceIri,RDF.TYPE,Dog.CLASS_IRI);
        if(statements.size() != 0){
            Dog dog = new Dog(instanceIri);
            if(nestingLevel > 0){
                nestingLevel--;
                setProperties(model, dog,nestingLevel);
            }
            return dog;
        }

        return null;
    }

    @Override
    public Collection<Dog> getAllInstancesFromModel(Model model,int nestingLevel)throws Exception{
        Model statements = model.filter(null,RDF.TYPE,Dog.CLASS_IRI);
        Collection<Dog> allInstances = new ArrayList<>();
        for(Statement statement:statements){
            Resource subject = statement.getSubject();
            if(subject.isIRI()){
                IRI iri = (IRI) subject;
                Dog dog = getInstanceFromModel(model,iri,nestingLevel);
                allInstances.add(dog);
            }
        }


        return allInstances;
    }

    @Override
    public void removeInstanceFromModel(Model model,IRI instanceIri) {
        Set<Value> rdfCollections = model.filter(instanceIri,null,null).objects();
        List<Value> listOfnodes = rdfCollections.stream().filter(o -> o.isBNode()).collect(Collectors.toList());
        for(Value node:listOfnodes){
            model.removeAll(getModelRDFCollection(model,(BNode)node));
            model.remove(instanceIri,null,(BNode)node);
        }
        model.remove(instanceIri,RDF.TYPE,Dog.CLASS_IRI);
        model.remove(instanceIri,null,null);
    }

    @Override
    public void updateInstanceInModel(Model model,Dog dog){
        Set<Value> rdfCollections = model.filter(dog.getIri(),null,null).objects();
        List<Value> listOfnodes = rdfCollections.stream().filter(o -> o.isBNode()).collect(Collectors.toList());
        for(Value node:listOfnodes){
            model.removeAll(getModelRDFCollection(model,(BNode)node));
            model.remove(dog.getIri(),null,(BNode)node);
        }

        Model statements = model.filter(dog.getIri(),null,null);
        statements.removeIf(event -> !event.getPredicate().equals(RDF.TYPE));
    }



}

File: onto-code-maker-result-tester/src/test/python/familyPython/serialization/DogSerialization.py

from rdflib.graph import URIRef, Literal,BNode,Collection
from rdflib.namespace import RDF
from Vocabulary import Vocabulary
from entities.Dog import Dog
from .SerializationModel import SerializationModel

class DogSerialization(SerializationModel):

    def addToModel(self,ontology, dog):
        ontology.add((dog.getIri(),RDF.type, dog.getClassIRI()))
        self.addPropertiesToModel(ontology,dog)


    def addPropertiesToModel(self,ontology, dog):
        pass

    def setProperties(self,ontology, dog, nestingLevel):
        pass

    def getInstanceFromModel(self,ontology, instanceIri,nestingLevel):
        if (instanceIri,RDF.type,Dog.CLASS_IRI) in ontology:
            dog =  Dog(instanceIri)
            if nestingLevel > 0:
                nestingLevel -= 1
                self.setProperties(ontology, dog,nestingLevel)
            return dog
        return None

    def getAllInstancesFromModel(self,ontology,nestingLevel):
        allInstances = set()
        for s, p, o in ontology.triples((None,RDF.type,Dog.CLASS_IRI)):
            if isinstance(s,URIRef) :
                dog = self.getInstanceFromModel(ontology,s,nestingLevel)
                allInstances.add(dog)
        return allInstances

    def removeInstanceFromModel(self,ontology, subject):
        ontology.remove((subject,RDF.type,Dog.CLASS_IRI))
        for statement in ontology.triples((subject,None,None)):
            ontology.remove(statement)



